com.microsoft.mrtk.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs

```diff
@@ -0,0 +1,312 @@
+                            jointTransform.rotation = Quaternion.LookRotation(jointPose.Position - joints[(int)TrackedHandJoint.Wrist].Position, jointPose.Up);
+                            break;
+                        default:
+                            // For all other bones, do a simple "IK" from the rigged joint to the joint data's position.
+                            jointTransform.rotation = Quaternion.LookRotation(jointPose.Position - jointTransform.position, joints[i-1].Up);
+                            break;
+                    }
+                }
+            }
+
+            // Compute and apply the adjusted scale of the hand.
+            // Over time, we'll grow or shrink the rigged hand
+            // to more accurately fit the actual size of the
+            // user's hand.
+
+            // How quickly the hand will grow or shrink
+            // to fit the user's hand size.
+            const float errorGainFactor = 0.1f;
+
+            // Reasonable minimum and maximum for how much
+            // the hand mesh is allowed to stretch to fit the user.
+            const float minScale = 0.8f;
+            const float maxScale = 1.1f;
+
+            // Apply.
+            handScale += -error * errorGainFactor;
+            handScale = Mathf.Clamp(handScale, minScale, maxScale);
+            transform.localScale = new Vector3(handNode == XRNode.LeftHand ? -handScale : handScale, handScale, handScale);
+
+            // Update the hand material based on selectedness value
+            UpdateHandMaterial();
+        }
+
+        // Computes the error between the rig's joint position and
+        // the user's joint position along the finger vector.
+        private float JointError(Vector3 armatureJointPosition, Vector3 userJointPosition, Vector3 fingerVector)
+        {
+            // The computed error between the rigged mesh's joints and the user's joints
+            // is essentially the distance between the mesh and user joints, projected
+            // along the forward axis of the finger itself; i.e., the "length error" of the finger.
+            return Vector3.Dot((armatureJointPosition - userJointPosition), fingerVector);
+        }
+
+        private bool ShouldRenderHand()
+        {
+            // If we're missing anything, don't render the hand.
+            if (handsSubsystem == null || wrist == null || handRenderer == null)
+            {
+                return false;
+            }
+
+            if (displaySubsystems.Count == 0)
+            {
+                SubsystemManager.GetSubsystems(displaySubsystems);
+            }
+
+            // Are we running on an XR display and it happens to be transparent?
+            // Probably shouldn't be showing rigged hands! (Users can
+            // specify showHandsOnTransparentDisplays if they disagree.)
+            if (displaySubsystems.Count > 0 &&
+                displaySubsystems[0].running &&
```

Comment:
Do we want to iterate through this list if there's more than one display subsystem to see if any are running?

Reply:
Thinking through this a bit more, I think this _should_ be okay. I don't think there are any platforms I'm aware of that would try to run more than one at a time, but it's possible!

Reply:
If there ever is a device with multiple display subsystems, I'll go back and update this :)